services/web/server/src/simcore_service_webserver/diagnostics_monitoring.py

```diff
@@ -23,22 +23,34 @@
 kCANCEL_COUNT = f"{__name__}.cancel_count"
 
 kCOLLECTOR_REGISTRY = f"{__name__}.collector_registry"
+kPROCESS_POOL_EXECUTOR = f"{__name__}.process_pool_executor.pool"
+
+
+def get_collector_registry(app: web.Application) -> CollectorRegistry:
+    return app[kCOLLECTOR_REGISTRY]
 
 
 async def metrics_handler(request: web.Request):
-    # TODO: prometheus_client.generate_latest blocking! -> Consider https://github.com/claws/aioprometheus
-    reg = request.app[kCOLLECTOR_REGISTRY]
-    resp = web.Response(body=prometheus_client.generate_latest(registry=reg))
-    resp.content_type = CONTENT_TYPE_LATEST
-    return resp
+    registry = get_collector_registry(request.app)
+
+    with concurrent.futures.ThreadPoolExecutor() as pool:
+        # NOTE: Cannot use ProcessPoolExecutor because registry is not pickable
```

Comment:
have you tried to see if request is `picklable` maybe that could do the trick and will let you fetch the registry in the new process?

Reply:
But if you are doing I/O it is better to use the ThreadPool instead of the ProcessPool, because that one has a higher overhead. Or to improve performance you could add a shared pool.

Reply:
Yes, that is already clear from the [executors documentation](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor). I would say is mostly CPU bound but I also believe here the problem is a bit more tricky because this function will behave differently depending on how [the metrics](https://prometheus.io/docs/concepts/metric_types/) are configured or how much data is accumulated. 

IN any case, this PR is simply a first essay to make the metrics call async by using the default executor. 
In order to further optimize it I would rather first tune [the metrics](https://prometheus.io/docs/concepts/metric_types/) dimensions .

Actually, based on the trials I have been doing, most of the metrics annotations come from files served for the front-end which means that the static-web-server will help making this call faster :-)

Reply:
> have you tried to see if request is `picklable` maybe that could do the trick and will let you fetch the registry in the new process?

but the registry is also in memory. How would you get it from the other process?